ws-server/src/server.rs

```diff
@@ -309,19 +314,28 @@ async fn background_task(
 	let stop_server2 = stop_server.clone();
 	let sink = MethodSink::new_with_limit(tx, max_response_body_size);
 
+	let mut submit_ping = tokio::time::interval(ping_interval);
+
 	middleware.on_connect();
 
 	// Send results back to the client.
 	tokio::spawn(async move {
 		while !stop_server2.shutdown_requested() {
-			if let Some(response) = rx.next().await {
-				// If websocket message send fail then terminate the connection.
-				if let Err(err) = send_ws_message(&mut sender, response).await {
-					tracing::warn!("WS send error: {}; terminate connection", err);
-					break;
+			select! {
```

Comment:
check if you could replace this `select!` with something that fetches the dropped future have a look https://github.com/paritytech/jsonrpsee/blob/master/core/src/server/rpc_module.rs#L904-#L955 for inspiration :)

:point_up: should be cancel-safe but would be good not use `select!` because it's a footgun.